Metalama.Framework/Code/TypeFactory.cs

```diff
@@ -72,15 +72,13 @@ public static IPointerType MakePointerType( this IType pointedType )
     /// Creates a nullable type from the current type. If the current type is already nullable, returns the current type.
     /// If the type is a value type, returns a <see cref="Nullable{T}"/> of this type.
     /// </summary>
-    public static T ToNullableType<T>( this T type )
-        where T : IType
+    public static IType ToNullableType( this IType type )
         => ((ICompilationInternal) type.Compilation).Factory.ConstructNullable( type, true );
 
     /// <summary>
     /// Returns the non-nullable type from the current type. If the current type is a non-nullable reference type, returns the current type.
     /// If the current type is a <see cref="Nullable{T}"/>, i.e. a nullable value type, returns the underlying type.
     /// </summary>
-    public static T ToNonNullableType<T>( this T type )
-        where T : IType
+    public static IType ToNonNullableType( this IType type )
         => ((ICompilationInternal) type.Compilation).Factory.ConstructNullable( type, false );
 }
```

Comment:
Add an overload `INamedType` -> `IType` with this remark:

> In case the current type represents a `Nullable<T>` where `T` is a value-type type parameter, this method returns  `ITypeParameter`. Otherwise, it returns an `INamedType`.